desktop/apps/auction2/routes.js

```diff
@@ -1,51 +1,85 @@
+import React from 'react'
+import SaleQuery from 'desktop/apps/auction2/utils/queries/sale'
+import auctionReducer, { initialState } from 'desktop/apps/auction2/reducers'
+import configureStore from 'desktop/apps/auction2/utils/configureStore'
+import footerItems from 'desktop/apps/auction2/utils/footerItems'
 import metaphysics from 'lib/metaphysics.coffee'
 import get from 'lodash.get'
-import renderTemplate from 'desktop/components/react/utils/render_template'
+import renderReactLayout from 'desktop/components/react/utils/render_react_layout'
+import u from 'updeep'
 import { getLiveAuctionUrl } from 'utils/domain/auctions/urls'
 
 export async function index (req, res, next) {
-  const { me } = await metaphysics({ query: MeQuery(req.params.id), req: req })
-  const { sale } = await metaphysics({ query: SaleQuery(req.params.id) })
-  const { articles } = await metaphysics({ query: ArticlesQuery(sale._id) })
+  const saleId = req.params.id
 
-  res.locals.sd.AUCTION = sale
+  try {
+    const {
+      sale
+    } = await metaphysics({
+      query: SaleQuery(saleId)
+    })
+
+    res.locals.sd.AUCTION = sale
+    fetchUser(req, res)
+
+    const store = configureStore(auctionReducer, {
+      auctionArtworks: u({
+        filterParams: {
+          sale_id: saleId
+        },
+        isClosed: sale.is_closed,
+        user: res.locals.sd.CURRENT_USER
+      }, initialState)
+    })
+
+    const [
+      { me },
+      { articles }
+    ] = await Promise.all([
+      metaphysics({ query: MeQuery(sale.id), req: req }),
+      metaphysics({ query: ArticlesQuery(sale._id) }),
+
+      // Hydrate store
+      store.dispatch(actions.fetchArtworksByFollowedArtists()),
+      store.dispatch(actions.fetchArtworks())
+    ])
+
+    const auctionModel = new Auction(sale)
+    const auctionArticles = new Articles(articles)
 
-  /**
-   * Ensure user model is in sync as updates to it during bidder registration
-   * route transitions will incidentally trigger a client-side reload.
-   * See: https://github.com/artsy/force/blob/master/desktop/lib/global_client_setup.coffee#L37
-   */
-  if (req.user) {
     try {
-      const user = await req.user.fetch()
-      res.locals.sd.CURRENT_USER = {
-        ...res.locals.sd.CURRENT_USER,
-        ...user
-      }
+      const layout = renderReactLayout({
+        basePath: res.app.get('views'),
+        blocks: {
+          head: 'Meta.jade',
+          body: (props) => <App store={store} {...props} />
+        },
+        locals: res.locals,
+        data: {
+          articles: auctionArticles,
+          assetPackage: 'auctions2',
+          auction: auctionModel,
+          auctionArtworks: store.getState().auctionArtworks,
+          bodyClass: 'body-header-fixed body-no-margins',
+          footerItems: footerItems,
+          isLiveOpen: auctionModel.isLiveOpen(),
+          liveAuctionUrl: getLiveAuctionUrl(auctionModel.get('id'), {
+            isLoggedIn: Boolean(me)
+          }),
+          me: me
```

Comment:
Also, per the client / server **temporary** `.jade` comment up above, another way to quickly create react components out of jade components is to add in a templates key and injected in as props:

```javascript
const layout = renderReactLayout({
  ...
  templates: {
     Foo: 'foo.jade',
     Bar: 'bar.jade'
  }
})
```